Automattic/wp-calypso
# Domains: Update how we handle domain suggestion vendors

#### Changes proposed in this Pull Request

- Extract logic to check if user on given locales.
- Use a function to get the vendor for suggestions. Move the a/b test assignment logic in the function.
- Split en/non-en traffic for suggestions in function.

#### Testing instructions

- Enable `'calypso:abtests'` debugging
- Set your user locale to `en`
- Visit `/start/domains`
- Make sure you get the right variation
- Set locale to something that's not `en`
- Make sure you get the right variation
- Now try incognito
- Set browser language to `en`
- Make sure you get the right variation
- Now change to non-en
- Make sure you get the right variation
